src/devices_models/devices/common/objective_function/market_bid.jl

```diff
@@ -14,40 +41,241 @@ _consider_startup_time_series(
             [get_variables(container)],
             VariableKey.([StartVariable, MULTI_START_VARIABLES...], [T])),
     )
-_consider_shutdown_time_series(
+
+_consider_parameter(
+    ::ShutdownCostParameter,
     container::OptimizationContainer,
     ::DeviceModel{T, D},
 ) where {T, D} =
     haskey(get_variables(container), VariableKey(StopVariable, T))
 
+_consider_parameter(
+    ::IncrementalCostAtMinParameter,
+    container::OptimizationContainer,
+    ::DeviceModel{T, D},
+) where {T, D} =
+    haskey(get_variables(container), VariableKey(OnVariable, T))
```

Comment:
we should have a method to do this without creating `VariableKey` objects. @m-bossart I think was looking at a similar application.